crates/ide/src/ide/assists/inline.rs

```diff
@@ -0,0 +1,152 @@
+use super::{AssistKind, AssistsCtx};
+use crate::def::{AstPtr, ResolveResult};
+use crate::TextEdit;
+use smol_str::{SmolStr, ToSmolStr};
+use syntax::ast::AstNode;
+use syntax::{ast, match_ast, SyntaxNode};
+
+pub(super) fn inline_from_reference(ctx: &mut AssistsCtx<'_>) -> Option<()> {
+    let file_id = ctx.frange.file_id;
+    let parse = ctx.db.parse(file_id);
+
+    let covering_node = ctx.covering_node::<ast::Ref>()?;
+    let covering_token = covering_node.token()?;
+    let ptr = AstPtr::new(covering_node.syntax());
+
+    let name_res = ctx.db.name_resolution(file_id);
+    let source_map = ctx.db.source_map(file_id);
+    let expr_id = source_map.expr_for_node(ptr)?;
+
+    let replacement = match name_res.get(expr_id)? {
+        &ResolveResult::Definition(name) => source_map.nodes_for_name(name).find_map(|ptr| {
```

Comment:
I think this single-arm match can be refactored to let-else instead. Not sure why clippy does not fire here.

Also that we should only provide the assist when `nodes_for_name` returns exactly one node, as I mentioned previously. The case of multiple nodes for a single name can only happen for `let a.b = 1; a.c = 2; in a`. Let's add negative tests for it to NOT fire either assist.